lib/src/revset.rs

```diff
@@ -405,6 +414,15 @@ impl RevsetExpression {
         })
     }
 
+    /// Descendants of `self`, including `self` up to `generation` back.
+    pub fn descendants_at(self: &Rc<RevsetExpression>, generation: u64) -> Rc<RevsetExpression> {
+        Rc::new(RevsetExpression::Descendants {
+            roots: self.clone(),
+            generation: (generation - 1)..generation,
```

Comment:
This should also be `generation..(generation + 1)`

fwiw, we might also need a constructor that takes generation range for `ancestors`/`descendants(x, n, m)`. I'm not sure if we should add a separate function for `n`-th generation, or reuse the range function.
https://github.com/martinvonz/jj/discussions/1984#discussioncomment-6643259